runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -26,6 +26,9 @@
 #include "runtime/JITServerAOTCache.hpp"
 #include "runtime/JITServerSharedROMClassCache.hpp"
 
+size_t JITServerAOTCacheMap::_cacheMaxBytes = 300*1024*2014;
```

Comment:
The default SCC size is 300 MB. @cjjdespres could you please determine how much AOT cache space is consumed by something like Acmeair?
Perhaps the AOT cache limit should also be a factor of the memory limit for the jitserver process.

Reply:
I remembered that I actually did these measurements already for the JITServer ATC paper, and the largest AOT cache size (measured the same way as here) I saw was ~130 MB for DayTrader7. Although this number does slowly grow over time as the "tail" of methods that rarely get compiled (and rarely get reused) accumulates in the cache. I think 300 MB is reasonable, but then ideally it should be the same as the default SCC size rather than a separate hard-coded constant. It's important that this number is per application, so we need to be very clear in the documentation that if multiple distinct applications use the same JITServer instance, the limit may need to be increased.

Reply:
I saw around 55MB with the AcmeAir benchmark after a few rounds of client connections and loads, but that's a lighter benchmark than DT7. Disabling the limit by default might be better, to keep the behaviour consistent with what it currently is.

Reply:
I am ok with having no limit by default, given that people typically set a limit for the entire process. What happens if a memory allocation for AOT cache data structures fails? Do we handle that case gracefully?

Reply:
Without the limit, AOT cache allocation failures are currently handled as `std::bad_alloc` exceptions which IIRC result in failed compilations. I think the same applies to everything else using the persistent allocator, i.e. client session caches, shared ROMClass cache, persistent IProfiler and CHTable data, etc. The only thing that is different about the AOT cache is that it always persists beyond client sessions, so it can grow unbounded.

I don't think that keeping existing behaviour is a concern at this point, thus setting a generous default limit is a good solution for now, but ideally we need to evict stale records and method from the AOT cache. I think the default SCC size (300 MB) is the minimum value that could be considered "generous". I don't think we should go above 2x of that. Also in my experience other memory usage (scratch and client caches) dominates the JITServer memory footprint with multiple concurrent clients anyway, which is where the AOT cache is most beneficial.

Reply:
Currently a `std::bad_alloc()` is thrown, which I think is caught in at least `TR::CompilationInfoPerThreadRemote::processEntry`.